src/offline/wallet.ts

```diff
@@ -0,0 +1,88 @@
+import {fromSeed} from 'bip32'
+import {mnemonicToSeedSync} from 'bip39'
+import {decode, encodeForSigning} from 'ripple-binary-codec'
+import {deriveKeypair, generateSeed, verify} from 'ripple-keypairs'
+import {signOffline} from '../transaction/sign'
+import {SignOptions} from '../transaction/types'
+
+export interface WalletOptions {
+  // The entropy to use to generate the seed.
+  entropy?: Uint8Array | number[]
+
+  // The digital signature algorithm to generate an address for. Can be `ecdsa-secp256k1` (default) or `ed25519`.
+  algorithm?: 'ecdsa-secp256k1' | 'ed25519'
+
+  // The path to derive a keypair (publicKey/privateKey) from a seed (that was converted from a mnemonic)
+  derivationPath?: string
+}
+
+/**
+ * A utility for deriving a wallet composed of a keypair (publicKey/privateKey).
+ * A wallet can be derived from either a seed, mnemnoic, or entropy (random number).
+ * It provides functionality to sign/verify transactions offline.
+ */
+class Wallet {
+  readonly publicKey: string
+  readonly privateKey: string
+  private static readonly defaultAlgorithm: 'ecdsa-secp256k1' | 'ed25519' =
+    'ed25519'
+  private static readonly defaultDerivationPath: string = "m/44'/144'/0'/0/0"
+
+  constructor(publicKey: string, privateKey: string) {
+    this.publicKey = publicKey
+    this.privateKey = privateKey
+  }
+
+  static walletFromSeed(seed: string, options: WalletOptions): Wallet {
+    return Wallet.deriveWallet(seed, options)
+  }
+
+  static walletFromMnemonic(mnemonic: string, options: WalletOptions): Wallet {
+    const seed = mnemonicToSeedSync(mnemonic)
+    const masterNode = fromSeed(seed)
+    const node = masterNode.derivePath(
+      options.derivationPath ?? Wallet.defaultDerivationPath
+    )
+    if (node.privateKey === undefined) {
+      return undefined
+    }
+
+    const publicKey = Wallet.hexFromBuffer(node.publicKey)
+    const privateKey = Wallet.hexFromBuffer(node.privateKey)
+    return new Wallet(publicKey, `00${privateKey}`)
```

Comment:
What does the `00` prefix on the `privateKey` do?

Reply:
I believe the `00` prefix is used to complete the derivation of `privateKey` from a seed (derived by a mnemonic). I followed the [same implementation xpring-common-js](https://github.com/xpring-eng/xpring-common-js/blob/1ff6c39e021fefeffa08d08ace14df74cd8b70da/src/XRP/wallet.ts#L140) does for this use case of wallet from mnemonic.